### ISSUE TITLE
Azure - Key Vault Create Events not reliably arriving

### ISSUE THREAD

**Issue by helayoty** [cloud-custodian/cloud-custodian#3343]
**Issue**

When trying to run policy to create an output blob storage when adding new key-vault, custodian fails to:
 1. add CreateEmail tag to the new key-vault
 2. create new blob storage to save the output for running the policy  

**Steps to Reproduce**

Attempt to run the following policy against your subscription (note you must have active app service plans)

`policies:
    - name: tag-key-vault-creator
      resource: azure.keyvault
      mode:
        type: azure-event-grid
        events: [{
            resourceProvider: 'Microsoft.KeyVault/vaults',
            event: 'write'
          }]
      filters:
        - "tag:CreatorEmail": null
      actions:
        - type: auto-tag-user
      `

by running the following command

`custodian run -s azure://mystorage.blob.core.windows.net/logs mypolicy.yml`

no error is shown although the blob storage and "CreateEmail" tag can't be found.

**Root Cause**

A bug found in event grid subscription

@erwelch @logachev  @aluong

**Comment by erwelch** [comment]
I've investigated this a bit and just adding a few more notes:
1. I was able to repro this inconsistently - sometimes it works as expected
2. When I can repro this if I go to the Event Subscription and make a small edit (like change casing) and save, then it relaunches the event subscription and picks up key vault write events
3. I have not been able to pick up a Key Vault create event, only edits. However the 'Microsoft.KeyVault/vaults/write' operation is documented to be for both create and update: https://docs.microsoft.com/en-us/azure/role-based-access-control/resource-provider-operations#microsoftkeyvault
